chainer/reporter.py

```diff
@@ -356,3 +363,16 @@ def make_statistics(self):
             stats[name + '.std'] = std
 
         return stats
+
+    def serialize(self, serializer):
+        if isinstance(serializer, serializer_module.Serializer):
+            serializer('_names', json.dumps(list(self._summaries.keys())))
+            for name, summary in six.iteritems(self._summaries):
+                if hasattr(summary, 'serialize'):
+                    summary.serialize(serializer['_summaries'][name])
+        else:
+            names = json.loads(serializer('_names', ''))
```

Comment:
I think we should clear `self._summaries` before starting deserialization.

Reply:
Why? All values are overwritten in deserialization.

Reply:
Because users may want to deserialize into existing `DictSummary` instance which has a different set of keys.

Reply:
OK